java-checks/src/main/java/org/sonar/java/checks/unused/UnusedLocalVariableCheck.java

```diff
@@ -61,6 +66,7 @@ public List<Tree.Kind> nodesToVisit() {
   public void visitNode(Tree tree) {
     if (tree.is(Tree.Kind.COMPILATION_UNIT)) {
       UNRESOLVED_IDENTIFIERS_VISITOR.check(tree);
+      SWITCH_PATTERN_VARIABLE_VISITOR.check(tree);
```

Comment:
Running 2 visitors on the full AST is time-consuming. Because `UnresolvedIdentifiersVisitor` and `SwitchPatternVariableVisitor` have the same life cycle, it should be possible to merger the two behaviors in a `VariableVisitor` that extends  `UnresolvedIdentifiersVisitor`.

Reply:
You are right. it's smarter. Fixed.